src/queries/label-text.js

```diff
@@ -12,30 +12,73 @@ import {
 } from './all-utils'
 import {queryAllByText} from './text'
 
+function getCombinations(labels, matcher) {
+  const combs = [[]]
+  const matching = []
+  for (const label of labels) {
+    const copy = [...combs] // See note below.
+    for (const prefix of copy) {
+      const combination = prefix.concat(label.textToMatch)
+      combs.push(combination)
+      if (matcher(combination.join(' '), label.node)) {
+        matching.push(label.node)
+      }
+    }
+  }
+  return matching
+}
+function queryAllLabels(container) {
+  return Array.from(container.querySelectorAll('label,input'))
```

Comment:
It occurs to me that an element can be labeled by *anything*, not just `label` and `input`. So while this may be solving for the input case, it doesn't solve for everything else 🤔

That said, I'm not sure how to handle "everything else." It seems to me that rather than searching for all possible elements that can label anything, we should be searching for everything that's labeled first, determine the contents of their label, and then return the elements whose label matches the query.

Reply:
I am not sure I am totally getting your point, are you telling that [this](https://github.com/testing-library/dom-testing-library/blob/732a8abf5560b13cfba976f5f801048e7018e2d4/src/queries/label-text.js#L20) loop runs only once inside the [outer](https://github.com/testing-library/dom-testing-library/blob/732a8abf5560b13cfba976f5f801048e7018e2d4/src/queries/label-text.js#L18) loop?

Reply:
For what it's worth, I think that searching for labelled items and then for their labels is more feasible than querying for all elements that could be labels. Like you said above!

Reply:
If you'd like to tack a whack at that, I think it would solve a lot of problems.

Reply:
For sure I would like to do it!